SConstruct

```diff
@@ -586,25 +611,127 @@ def list_sources(base, suffixes):
+    append_sources(result, *list_sources('src/ripple/legacy', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/net', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/overlay', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/peerfinder', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/protocol', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/shamap', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/nodestore', '.cpp'),
+        CPPPATH=[
+            'src/rocksdb2/include',
+            'src/snappy/snappy',
+            'src/snappy/config',
+        ])
+
+    result += get_soci_sources('classic')
+    return result
+
+def get_unity_sources():
+    result = []
+    append_sources(
+        result,
+        'src/ripple/unity/app.cpp',
+        'src/ripple/unity/app1.cpp',
+        'src/ripple/unity/app2.cpp',
+        'src/ripple/unity/app3.cpp',
+        'src/ripple/unity/app4.cpp',
+        'src/ripple/unity/app5.cpp',
+        'src/ripple/unity/app6.cpp',
+        'src/ripple/unity/app7.cpp',
+        'src/ripple/unity/app8.cpp',
+        'src/ripple/unity/app9.cpp',
+        'src/ripple/unity/core.cpp',
+        'src/ripple/unity/basics.cpp',
+        'src/ripple/unity/crypto.cpp',
+        'src/ripple/unity/net.cpp',
+        'src/ripple/unity/overlay.cpp',
+        'src/ripple/unity/peerfinder.cpp',
+        'src/ripple/unity/json.cpp',
+        'src/ripple/unity/protocol.cpp',
+        'src/ripple/unity/shamap.cpp',
+        'src/ripple/unity/legacy.cpp',
+    )
+
+    result += get_soci_sources('unity')
+
+    append_sources(
+        result,
+        'src/ripple/unity/nodestore.cpp',
+        CPPPATH=[
+            'src/rocksdb2/include',
+            'src/snappy/snappy',
+            'src/snappy/config',
+        ])
+
+    return result
 
 # Declare the targets
 aliases = collections.defaultdict(list)
 msvc_configs = []
 
+def should_prepare_target (cl_target,
+                           style, toolchain, variant):
+    if not cl_target:
+        # default target
+        return (style == default_tu_style and
+                toolchain == default_toolchain and
+                variant == default_variant)
+    if 'vcxproj' in cl_target:
+        return toolchain == 'msvc'
+    s = cl_target.split('.')
+    if style == 'unity' and 'nounity' in s:
+        return False
+    if len(s) == 1:
+        return 'all' in cl_target or variant in cl_target or toolchain in cl_target
+    if len(s) == 2 or len(s) == 3:
+        return s[0] == toolchain and s[1] == variant
+
+    return True # A target we don't know about, better prepare to build it
+
+def should_prepare_targets (style, toolchain, variant):
+    if not COMMAND_LINE_TARGETS:
+        return should_prepare_target (None, style, toolchain, variant)
+    for t in COMMAND_LINE_TARGETS:
+        if should_prepare_target (t, style, toolchain, variant):
+            return True
+        return False
```

Comment:
OOPS... I think line 727 got indented one column too far, because the loop can't ever execute more than once.

But actually - you can just _delete_ it.  All Python functions return a value - if you don't explicitly call `return` they return `None`, which is "falsey" (https://www.udacity.com/wiki/cs258/truthiness-in-python)